all: Update op-geth dependency, based on geth v1.16.1

0x00101010 opened the pull request:
<!--
Contributions welcome! See https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md
-->

**Description**

Updates op-geth dependency to include upstream geth v1.16.1 changes from https://github.com/ethereum-optimism/op-geth/pull/638

<!--
A clear and concise description of the features you're adding in this pull request.
-->

**Tests**

<!--
Please describe any tests you've added. If you've added no tests, or left important behavior untested, please explain why not.
-->

**Additional context**

<!--
Add any other context about the problem you're solving.
-->

**Metadata**

<!-- 
Include a link to any github issues that this may close in the following form:
- Fixes #[Link to Issue]
-->

0x00101010 committed:
Upgrade op-geth with upstream v1.16.1 version

0x00101010 committed:
Update to fix linters

0x00101010 committed:
go mod tidy

0x00101010 committed:
Fix part of tests

sebastianst commented on op-program/client/l2/test/miner.go:
```diff
@@ -40,6 +40,7 @@ func NewMiner(t *testing.T, logger log.Logger, isthmusTime uint64) (*Miner, *cor
 	config.HoloceneTime = &zero
 	config.IsthmusTime = &isthmusTime
 	config.PragueTime = &isthmusTime
+	config.OsakaTime = &isthmusTime
```
why would we enable Osaka with Isthmus?

sebastianst replied:
@0x00101010 I'm reverting this, let me know if this was intentional

sebastianst replied:
I see, it was done because the default config now sets it to zero, which is incompatible. Fixed it by actually deactivating osaka for this test fully for now. We'll need to change it later when we do proper L1 Osaka prep.

sebastianst commented on op-wheel/cheat/cheat.go:
```diff
@@ -48,7 +47,11 @@ func OpenGethRawDB(dataDirPath string, readOnly bool) (ethdb.Database, error) {
-	db, err := rawdb.NewDatabaseWithFreezer(kvs, filepath.Join(dataDirPath, "ancient"), "", readOnly)
+	db, err := rawdb.Open(kvs, rawdb.OpenOptions{
+		Ancient:          filepath.Join(dataDirPath, "ancient"),
+		MetricsNamespace: "",
```
```suggestion
```
nit, can omit empty default

opgitgovernance commented:
This pr has been automatically marked as stale and will be closed in 5 days if no updates

sebastianst committed:
Update op-geth dependency

sebastianst committed:
don't enable osaka in op-program test miner

sebastianst committed:
Merge branch 'develop' into op-geth-upgrade

sebastianst committed:
Bump op-geth version

sebastianst committed:
Disable Osaka in op-program test miner

Inphi committed:
op-program: Refresh cannon compat report (#17071)

This is done for an incoming op-geth dependency update.
The new vm-compat findings highlight a new control flow path where the
flock syscall could be used. Similar control flow paths already exist in
the existing compatibility report. And these are benign for the same
reason; because they're only reachable by a program that uses a real
file-based ethdb.Database. The op-program uses a memory-based
ethdb.Database implementation to ensure that no file-related operations
occur during its execution.

The other flagged (WARNING) syscall use is related to the above flock
operation. This occurs when the go runtime issues epoll_ctl syscalls to
synchronize the file lock. Since we know that file operations won't
occur, this syscall use is also unreachable.

sebastianst commented:
Remaining failing tests are due to in-block state prefetching (https://github.com/ethereum/go-ethereum/pull/31557). The L1Cost fn reads the parameters from state each time the block num changes, which leads to concurrent map write races when many transactions are executed at once. Checking how we can better initialize the cost fn, avoiding races.

sebastianst committed:
Bump op-geth version

sebastianst committed:
op-service/eth: fix validationError string pointer reference

sebastianst commented:
The L1Cost issue is solved.

However, changes unrelated to the upstream merge are now causing other tests to fail, see https://github.com/ethereum-optimism/optimism/pull/17154 where I isolate the changes from https://github.com/ethereum-optimism/op-geth/commit/1b010f9e8ebad62e4b722186f8e2f710b756e5e2 to see which remaining test failures are caused by the refactor of the engine API checks.

sebastianst committed:
op-service/eth: fix exec payload creation from block

Execution payloads must only set the withdrawals root post-Isthmus.
The block's withdrawals root is set during assembly pre-Isthmus.

sebastianst committed:
Bump op-geth version

sebastianst committed:
Merge remote-tracking branch 'origin/develop' into op-geth-upgrade

sebastianst commented:
https://github.com/ethereum-optimism/optimism/pull/17154 fixed the tests related to engine api checks. Now only a few failing tests remain in `github.com/ethereum-optimism/optimism/op-e2e/actions/helpers`...

sebastianst committed:
Bump op-geth version (catch nil chainView)

protolambda commented on go.mod:
```diff
@@ -301,12 +299,11 @@ require (
 )
 
-replace github.com/ethereum/go-ethereum => github.com/ethereum-optimism/op-geth v1.101511.2-0.20250820165240-085b529eefe9
+replace github.com/ethereum/go-ethereum => github.com/ethereum-optimism/op-geth v1.101511.2-0.20250822133336-3edab82e9ca8
```
Checked: this doesn't include the very latest commit, but an older version of it, that is only formatted differently. Not a problem for testing.
We will need to update this to the op-geth `optimism` branch commit once the op-geth PR is merged anyway.

sebastianst replied:
Yes, I was just brushing up commit messages, no content change

protolambda reviewed (commented):
LGTM! (Not approving yet; we need to wait for op-geth merge + `go.mod` update before merging this)

sebastianst committed:
Bump op-geth dependency to v1.101601.0-rc.1
